lib/remote_persistent_term/fetcher/http.ex

```diff
@@ -3,20 +3,46 @@ defmodule RemotePersistentTerm.Fetcher.Http do
   A Fetcher implementation for HTTP.
   """
   require Logger
+  alias RemotePersistentTerm.Fetcher.Http.Cache
 
   @behaviour RemotePersistentTerm.Fetcher
 
   @type t :: %__MODULE__{
-          url: String.t()
+          url: String.t(),
+          http_cache?: boolean(),
+          min_refresh_interval_ms: pos_integer()
         }
 
-  defstruct [:url]
+  defstruct [:url, :http_cache?, :min_refresh_interval_ms]
 
   @opts_schema [
     url: [
       type: :string,
       required: true,
       doc: "The url from which the remote term is downloaded."
+    ],
+    http_cache: [
+      type: :keyword_list,
+      doc: "Configuration options for the HTTP Caching spec.",
+      default: [],
+      keys: [
+        enabled?: [
+          type: :boolean,
+          default: false,
+          doc: """
+          If true, the HTTP Caching spec will be used to schedule the next download.
+          The user should avoid setting both this value to true and `RemotePersistentTerm.refresh_interval` to a value.
```

Comment:
```suggestion
          If true, the HTTP Caching spec will be used to schedule the next download and `:refresh_interval` can be omitted.
```

They can still do both if they really wanted to, right?